### DISCUSSION TITLE
Full path is not saved to media when I implement django-tenants in my django project

### DISCUSSION THREAD

**Discussion by wdavilav** [django-tenants/django-tenants#968] [General]
I am implementing multitenant in a django project with PostgreSQL for this I implemented a library called **django-tenants** but I have a small problem what I wanted was to save the media files for each scheme in a folder with the same name of the scheme for it In the official documentation it indicated that I had to put this configuration in my settings.py.

```
    DEFAULT_FILE_STORAGE = 'django_tenants.files.storage.TenantFileSystemStorage'
    
    MULTITENANT_RELATIVE_MEDIA_ROOT = "" 
```

I have an example model

```
    class Product(models.Model):
        name = models.CharField(max_length=150, verbose_name='Nombre')
        image = models.ImageField(upload_to='product/%Y/%m/%d', null=True, blank=True, verbose_name='Imagen')
```

But when I save a record, a folder with the name of the current scheme is created in the media along with the image, but in my database the image is saved without the name of the scheme, that is.

```
    You save /product/2023/08/30/imagen.png
    
   Instead of esquema1/product/2023/08/30/imagen.png
```

How could I solve this? I have searched for different ways but have not been able to achieve it.

**Comment by RodriGoncaDeeployer** [comment]
I did created a setting in my settings.py like `TENANT_STORAGE_FOLDER = 'tenant_storage'

And then in the models for files i did this:

`class AbstractFile(models.Model):

    APP_FOLDER = ''

    class Meta:
        abstract = True
        ordering = ['-created_at']

    def _get_file_path(self, filename):
        ext = filename.split('.')[-1]
        filename = filename.replace(filename.split('.')[-1], "")
        filename = "%s.%s" % (slugify(filename), ext)
        slug = self.tenant.slug if self.tenant.slug != '' else self.tenant.name.lower()
        return os.path.join(settings.TENANT_STORAGE_FOLDER, slug, self.APP_FOLDER, filename)

    id           = models.UUIDField(default=uuid.uuid4, primary_key=True, editable=False, verbose_name=_("ID"))
    name         = models.CharField(max_length=255, blank=True, null=True, verbose_name=_("Name"))
    tenant       = models.ForeignKey(Tenant, null=False, blank=False, on_delete=models.CASCADE, verbose_name="Tenant")
    file         = models.FileField(upload_to=_get_file_path,
                                    max_length=254,
                                    blank=False,
                                    null=False,
                                    validators=[FileExtensionValidator(allowed_extensions=settings.ALLOWED_DOCUMENTS_EXTENSIONS)],
                                    verbose_name=_("File Path"))`

I'm using Azure Storage Account as storage backend, I tried to use a different container for each tenant but could not figure it out.
But this aproach is working well. I hope it helps.
